pkg/diag/validator/pod.go

```diff
@@ -283,8 +284,11 @@ func (p *podStatus) String() string {
 }
 
 func extractErrorMessageFromWaitingContainerStatus(po *v1.Pod, c v1.ContainerStatus) (proto.StatusCode, []string, error) {
-	switch c.State.Waiting.Reason {
 	// Extract meaning full error out of container statuses.
+	switch c.State.Waiting.Reason {
+	case podInitializing:
+		// container is waiting to run
+		return proto.StatusCode_STATUSCHECK_SUCCESS, nil, nil
```

Comment:
Reading this -> https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#detailed-behavior
I am wondering if we should wait for all init containers to run successfully and pod to be running within in `statusCheckDeadline` or return success here like you have.